Resideo_eshopping/resideo_eshopping/lib/Screens/screen3.dart

```diff
@@ -0,0 +1,185 @@
+                RegExp regex = new RegExp(pattern);
+                if (!regex.hasMatch(val))
+                  return 'Enter Valid Phone Number';
+                else
+                  return null;
+              },
+              keyboardType: TextInputType.number,
+            ),
+            TextFormField(
+              maxLines: 3,
+              decoration: const InputDecoration(
+                icon: const Icon(
+                  FontAwesomeIcons.home,
+                  color: Color(0xff11b719),
+                ),
+                labelText: 'Address',
+              ),
+              validator: (val) {
+                if (val.length == 0)
+                  return 'Address Field should not be empty';
+                else
+                  return null;
+              },
+            ),
+            TextFormField(
+              decoration: const InputDecoration(
+                icon: const Icon(
+                  FontAwesomeIcons.mapMarker,
+                  color: Color(0xff11b719),
+                ),
+                hintText: 'Ex: 500000',
+                labelText: 'Zip Code',
+              ),
+              validator: (val) {
+                Pattern pattern = r'^[1-9][0-9]{5}$';
+                RegExp regex = new RegExp(pattern);
+                if (!regex.hasMatch(val))
+                  return 'Enter Valid ZipCode';
+                else
+                  return null;
+              },
+              keyboardType: TextInputType.number,
+            ),
+            Row(
+              crossAxisAlignment: CrossAxisAlignment.start,
+              children: <Widget>[
+                Container(
+                  margin: const EdgeInsets.all(20.0),
+                  child: Text(
+                    'Estimated Price: ',
+                    style: TextStyle(
+                        fontWeight: FontWeight.bold,
+                        fontSize: 20,
+                        color: Color(0xff11b719)),
+                  ),
+                ),
+                Container(
+                  margin: const EdgeInsets.all(20.0),
+                  child: Text(
+                    'Rs. $estimatedCost',
+                    style: TextStyle(fontWeight: FontWeight.bold, fontSize: 20),
+                  ),
+                ),
+              ],
+            ),
+            SizedBox(
+              height: 200.0,
+            ),
+            Column(
+              crossAxisAlignment: CrossAxisAlignment.stretch,
+              children: <Widget>[
+                RaisedButton(
+                    color: Color(0xff11b719),
+                    textColor: Colors.white,
+                    child: Padding(
+                        padding: EdgeInsets.all(10.0),
+                        child: Row(
+                          mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+                          children: <Widget>[
+                            Text("Proceed", style: TextStyle(fontSize: 24.0)),
```

Comment:
On Clicking proceed button, please show the alert dialog only if the user enter all the fields with proper values

Reply:
done